lib/jobs/docker-job.js

```diff
@@ -0,0 +1,230 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var stream = require('stream');
+
+module.exports = dockerJobFactory;
+
+di.annotate(dockerJobFactory, new di.Provide('Job.Docker'));
+di.annotate(dockerJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Constants',
+    'Services.Waterline',
+    'Services.Messenger',
+    'Docker'
+));
+
+function dockerJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Constants,
+    waterline,
+    messenger,
+    Docker
+) {
+    var logger = Logger.initialize(dockerJobFactory);
+
+    function DockerJob(options, context, taskId) {
+        BaseJob.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+    }
+    util.inherits(DockerJob, BaseJob);
+
+    DockerJob.prototype._run = function run() {
+        var self = this;
+
+        return waterline.nodes.needByIdentifier(self.nodeId)
+            .then(function (node) {
+                if (node && node.type === Constants.NodeTypes.ComputerContainer) {
```

Comment:
Should we add ComputerContainer to list of NodeTypes here:
https://github.com/RackHD/on-core/blob/master/lib/common/constants.js#L312